sherpa/astro/io/crates_backend.py

```diff
@@ -106,52 +113,62 @@ def get_filename_from_dmsyntax(filename, blockname=None):
     return arg
 
 
-def _try_hdr_key(crate, name, dtype=str):
-    """
-    checked for new crates
-    """
-    name = name.upper().strip()
-    if not crate.key_exists(name):
-        return None
-
-    return _try_key(crate, name, dtype)
+def _try_key(crate, name, dtype=str):
+    """Access the key from the crate returning None if it does not exist.
 
+    There's no way to differentiate between a key that does not exist
+    and a key that is set but has been set to a value of None (which
+    is unlikely as the ASCII and FITS serializations do not support
+    this).
 
-def _try_key(crate, name, dtype=str):
-    """
-    checked for new crates
-    """
+    Parameters
+    ----------
+    crate : TABLECrate or IMAGECrate
+    name : str
+       The name is case insensitive.
+    dtype
 
-    if not crate.key_exists(name):
-        return None
+    Returns
+    -------
+    value
+       The key value or None.
 
-    key = pycrates.get_key(crate, name)
+    """
 
+    # The get_key method will raise an IndexError if sent an integer
+    # outside the valid range, but here name is a string, where it
+    # returns None. This is inconsistent behaviour in crates circa
+    # CIAO 4.15.
+    #
+    key = crate.get_key(name)
     if key is None:
         return None
 
-    key = key.value
+    value = key.value
 
-    if str(key).find('none') != -1:
+    # Note that this check seems a bit too permissive, since a value
+    # of "This key is not set to none" will be matched.
```

Comment:
but since key in FITS are limited to 8 characters, we are probably OK in practice.

Reply:
a) you can actually have a longer key, it just needs "shenanighans"

b) the use of the name "key" for the argument here is a bit confusing and it actually refers to the value being stored, not the name of the key (and crates has a concept of key which contains all the relevant values). I actually have it renamed to `value` by the end of this PR